### PR TITLE
[Debt] Bump environment to PHP 8.3

### LINKED ISSUE #11819: ♻️ Upgrade app services to PHP 8.3
## ♻️ Debt/Refactor

PHP 8.3 is available as a setting for our app services.  Let's upgrade.

## ✅ Acceptance Criteria

- [x] App services bumped to 8.3
- [x] Azure build pipeline bumped to 8.3
- [x] GH workflows bumped to 8.3
- [x] Local webserver image bumped to 8.3
- [x] Maintenance container bumped to 8.3
- [x] Documentation updated

## 🛑 Blockers

```[tasklist]
### Blocked By
```

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #11819 

## 👋 Introduction

Bumps the environment to PHP 8.3.

### Locally

1. Bring your docker compose network down and rebuild the images
  - `docker compose build webserver`
  - `docker compose build maintenance`
2. Rebuild the `api` project

- [ ] The app still runs exactly as before

### Azure
- deploy app service with php 8.3
- deploy this branch into the app service

- [ ] The app still runs exactly as before

## 🚚 Deployment

- Set phpVersion in IaC to 8.3
  - [X] dev app servce
  - [ ] uat app service
  - [ ] prod app service
  - [ ] set default value in VERTICAL file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ hydrogen-logs
 .jekyll-cache
 .turbo
 .env
+hostingstart.html
```
Comment (petertgiles) [inline .gitignore:11]: Annoyingly, the docker image creates this file on startup.

File: infrastructure/webserver.Dockerfile [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 
 # All images: https://mcr.microsoft.com/v2/appsvc/php/tags/list
-FROM mcr.microsoft.com/appsvc/php:8.2-fpm-xdebug_20230908.3.tuxprod
+FROM mcr.microsoft.com/appsvc/php:8.3-fpm-xdebug_20241021.7.tuxprod
 
 RUN echo 'memory_limit=256M' >> /usr/local/etc/php/conf.d/php.ini
 
 RUN apt-get update \
     && apt-get install --yes --no-install-recommends supervisor cron postgresql-client \
-    && apt-get --yes autoremove \
```
Comment (petertgiles) [inline infrastructure/webserver.Dockerfile:10]: This autoremove was deleting a needed libargon2.so.1 file.

**Review by petertgiles — COMMENTED** [review]
A few notes.

File: api/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM php:8.2.9-apache-bullseye as base
+FROM php:8.3.9-apache-bullseye as base
```
Comment (esizer) [inline api/Dockerfile:1]: Do we want to stick with this path release? It looks like they are currently on `8.3.13`.

https://www.php.net/releases/

**Review by esizer — COMMENTED** [review]
Works well testing locally, will need someone else to test in dev. I just had a question about the patch release specified (I think it is just in the dockerfile and composer right now)

File: api/Dockerfile (hunk 3 above)
Comment (petertgiles) [reply api/Dockerfile:1]: I chose 8.2.9 because that matches the PHP version in our Azure app service image.

**Review by brindasasi — APPROVED** [review]
works fine on Dev !

**Pull request merged**